[i2c,doc] DV documentation review (#18088)
lowRISC/opentitan: hw/ip/i2c/dv/README.md

* [csr_utils_pkg](../../../dv/sv/csr_utils/README.md)

### Global types & methods
All common types and methods defined at the package level can be found in [`i2c_env_pkg`](https://github.com/lowRISC/opentitan/blob/master/hw/ip/i2c/dv/env/i2c_env_pkg.sv).
Some of them are imported from [`i2c_reg_pkg`](https://github.com/lowRISC/opentitan/blob/master/hw/ip/i2c/rtl/i2c_reg_pkg.sv).

### TL_agent
I2C instantiates (already handled in CIP base env) [tl_agent](../../../dv/sv/tl_agent/README.md)
which provides the ability to drive and independently monitor random traffic via
TL host interface into I2C device.

### I2C agent

> I2C agent is configured to work device mode and implemented as [reactive agent](https://www.verilab.com/files/mastering_reactive_slaves.pdf).

Reviewer: shouldn't this say "to work **in** device mode"?

All test sequences are extended from `i2c_base_vseq`.
It provides commonly used handles, variables, functions and tasks that the test sequences can simple use / call.
Some of the most commonly used tasks / functions are as follows:
* `i2c_init`: Setup the control register, identifier and clear the FIFOs.
* `wait_{host|target}_for_idle`.

> There's also another test sequence which multiple other tests use as a base and that is `i2c_rx_tx_vseq`.

Reviewer 2: I think this needs to be rephrased, specially the end. Maybe:
```suggestion
There's also the test sequence `i2c_rx_tx_vseq` which multiple other tests use as a base.
```

Reviewer 3: How about...
```suggestion
Another test sequence used as a base for other sequences is `i2c_rx_tx_vseq`.
```

Author: Thanks, made this change.

> This sequence is useful for tests that want to generate receive or transmit packets.

Reviewer 2: ```suggestion
This sequence is useful for tests that want to generate, receive or transmit packets.
```

Author: Thanks, fixed.

#### Functional coverage
To ensure high quality constrained random stimulus, it is necessary to develop a functional coverage model.
Please see the [testplan](../data/i2c_testplan.hjson) for details of the covergroups.

### Self-checking strategy
#### Scoreboard
The `i2c_scoreboard` is primarily used for end to end checking.

> It has a TLUL interface to monitor interactions with the registers.

Reviewer 2: I mostly see this way across the documentation.
```suggestion
It has a TL-UL interface to monitor interactions with the registers.
```

Author: Thanks, changed.

It also creates the following queues to monitor I2C packets:
* `rd_item_fifo`: for read items.
* `wr_item_fifo`: for write items.

#### Assertions

> * TLUL assertions: The `dv/sva/i2c_bind.sv` binds the `tlul_assert` [assertions](../../tlul/doc/TlulProtocolChecker.md) to the IP to ensure TileLink interface protocol compliance.

Author: Good point, I've made this change too.
